[TBL] Call collex directly for reports

JamesGardiner opened the pull request:
# Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
Removes the call to backstage for getting chasing reports.

# What has changed
<!--- What code changes has been made -->
<!--- Has there been any refactoring -->
<!--- What tests have been written -->
Changes the `download_report` method in collection_exercise_controller.py to hit the collection exercise endpoint directly.

# How to test?
<!--- Describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to see how your change affects other areas of the code, etc. -->
<!--- Are there any automated tests that mean changes don't need to be manually changed -->
Run the unit tests, make sure they pass.
Run the unit tests, make sure they pass.
Tail the logs for backstage, collection-exercise service. Hit the `response_chasing/<ce_id>/<survey_id>` endpoint using `curl` or Postman. Make sure backstage doesn't get the call, but collection-exercise service does.

jcox-dev commented on response_operations_ui/controllers/collection_exercise_controllers.py:
```diff
@@ -11,198 +11,311 @@
+        case_group["caseGroupStatus"]
+        for case_group in case_groups
+        if case_group["collectionExerciseId"] == collection_exercise_id
+    )
```
👍

jcox-dev commented on config.py:
```diff
@@ -44,6 +44,11 @@ class Config(object):
+    RM_REPORT_SERVICE_HOST = os.getenv('RM_REPORT_SERVICE_HOST', 'localhost')
+    RM_REPORT_SERVICE_PORT = os.getenv('RM_REPORT_SERVICE_PORT', 8084)
+    RM_REPORT_SERVICE_PROTOCOL = os.getenv('RM_REPORT_SERVICE_PROTOCOL', 'http')
+    RM_REPORT_SERVICE = f'{RM_REPORT_SERVICE_PROTOCOL}://{RM_REPORT_SERVICE_HOST}:{RM_REPORT_SERVICE_PORT}/'
```
Should be using the new NAME_URL service form with defaults (only?) in the DevelopmentConfig.

Do we also need separate auth details for the report service?

benjefferies replied:
I presume to see the env vars to be condensed

JamesGardiner commented:
@benjefferies and @jcox-dev - I've addressed your comments. There is no concept of basic auth yet for report service. This should be added by @AdamHawtin and co when it is implemented.

benjefferies commented on config.py:
```diff
@@ -44,6 +44,11 @@ class Config(object):
+    REPORT_HOST = os.getenv('REPORT_HOST')
+    REPORT_PORT = os.getenv('REPORT_PORT')
+    REPORT_PROTOCOL = os.getenv('REPORT_PROTOCOL')
+    REPORT_URL = f'{REPORT_PROTOCOL}://{REPORT_HOST}:{REPORT_PORT}/'
```
@JamesGardiner I was thinking a bit more along the lines of how it's done for SECURE_MESSAGE_URL

jcox-dev commented:
![](https://github.trello.services/images/mini-trello-icon.png) [response-operations-ui should directly call other services (not via backstage)](https://trello.com/c/0lPdzRbt/144-response-operations-ui-should-directly-call-other-services-not-via-backstage)

jcox-dev commented on config.py:
```diff
@@ -44,6 +44,11 @@ class Config(object):
     PARTY_PASSWORD = os.getenv('PARTY_PASSWORD')
     PARTY_AUTH = (PARTY_USERNAME, PARTY_PASSWORD)
 
+    REPORT_HOST = os.getenv('REPORT_HOST')
```
@JamesGardiner still need to update the standard Config to something like:

```python
REPORT_URL = os.getenv('REPORT_URL')
```

JamesGardiner commented:
Cheers @benjefferies
